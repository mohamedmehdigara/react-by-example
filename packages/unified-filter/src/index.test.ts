import remark from 'remark'
import { Processor } from 'unified'
import unifiedFilter from '.'

const fixtures = {
  simple: `\
# Hello world!

## Heading level 2

\`\`\`js
console.log('markdown')
\`\`\`

> blockquote

\`\`\`html
<p>The quick brown fox jumped over the fence.</p>
\`\`\`
`,
}

describe('unifiedFilter', () => {
  let processor: Processor

  beforeEach(() => {
    processor = remark().use(unifiedFilter, {
      filter: (n) => n.type !== 'code',
    })
  })

  it.each(Object.entries(fixtures))(
    'should filter nodes "%s"',
    async (_name, markdown) => {
      const result = await processor.process(markdown)
      expect(result.toString()).toMatchSnapshot()
    },
  )
})
