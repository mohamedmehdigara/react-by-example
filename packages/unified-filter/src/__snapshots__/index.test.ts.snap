// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unifiedFilter should filter nodes "simple" 1`] = `
"# Hello world!

## Heading level 2

> blockquote
"
`;
